mitosheet/mitosheet/tests/ai/test_recon.py

```diff
@@ -320,138 +397,200 @@ def test_exec_for_recon(code, dfs, recon_data):
     (
         pd.DataFrame({'A': [123], True: [456], 3: [789]}),
         pd.DataFrame({'A': [123], False: [456], 4: [789]}),
         {
-            'created_columns': [],
-            'deleted_columns': [],
-            'modified_columns': [],
-            'renamed_columns': {True: False, 3: 4}
+            'column_recon': {
+                'created_columns': [],
+                'deleted_columns': [],
+                'modified_columns': [],
+                'renamed_columns': {True: False, 3: 4}
+            }, 
+            'num_added_or_removed_rows': 0
         }
     ),
     # Reorder them
     (
         pd.DataFrame({'A': [123], True: [456], 3: [789]}),
         pd.DataFrame({True: [456], 3: [789], 'A': [123]}),
         {
-            'created_columns': [],
-            'deleted_columns': [],
-            'modified_columns': [],
-            'renamed_columns': {}
+            'column_recon': {
+                'created_columns': [],
+                'deleted_columns': [],
+                'modified_columns': [],
+                'renamed_columns': {}
+            }, 
+            'num_added_or_removed_rows': 0
         }
     ),
     # Modify a column
     (
         pd.DataFrame({'A': [123], True: [456], 3: [789]}),
         pd.DataFrame({'A': [1], True: [456], 3: [789]}),
         {
-            'created_columns': [],
-            'deleted_columns': [],
-            'modified_columns': ['A'],
-            'renamed_columns': {}
+            'column_recon': {
+                'created_columns': [],
+                'deleted_columns': [],
+                'modified_columns': ['A'],
+                'renamed_columns': {}
+            }, 
+            'num_added_or_removed_rows': 0
         }
     ),
     # Modify multiple columns
     (
         pd.DataFrame({'A': [123], True: [456], 3: [789]}),
         pd.DataFrame({'A': [1], True: [4], 3: [2]}),
         {
-            'created_columns': [],
-            'deleted_columns': [],
-            'modified_columns': ['A', True, 3],
-            'renamed_columns': {}
+            'column_recon': {
+                'created_columns': [],
+                'deleted_columns': [],
+                'modified_columns': ['A', True, 3],
+                'renamed_columns': {}
+            }, 
+            'num_added_or_removed_rows': 0
         }
     ),
     # No modify with NaN
     (
         pd.DataFrame({'A': [None], True: [None], 3: [789]}),
         pd.DataFrame({'A': [None], True: [None], 3: [789]}),
         {
-            'created_columns': [],
-            'deleted_columns': [],
-            'modified_columns': [],
-            'renamed_columns': {}
+            'column_recon': {
+                'created_columns': [],
+                'deleted_columns': [],
+                'modified_columns': [],
+                'renamed_columns': {}
+            }, 
+            'num_added_or_removed_rows': 0
+        }
+    ),
+    # Add rows, only sees rows as added and nothing modified
```

Comment:
If a user adds/deletes rows and adds/deletes a dataframe we should show them that. Can we add a test?